solutions/banking/main.tf

```diff
@@ -15,51 +15,51 @@ resource "random_string" "unique_identifier" {
 data "ibm_iam_auth_token" "tokendata" {}
 
 # Resource group - create if it doesn't exist
-# module "resource_group" {
-#   source  = "terraform-ibm-modules/resource-group/ibm"
-#   version = "1.1.5"
-#   # If an existing resource group is not set (null), then create a new one
-#   resource_group_name          = var.resource_group_name == null ? local.unique_identifier : null
-#   existing_resource_group_name = var.resource_group_name
-# }
+module "resource_group" {
+  source  = "terraform-ibm-modules/resource-group/ibm"
+  version = "1.1.5"
+  # If an existing resource group is not set (null), then create a new one
+  resource_group_name          = var.resource_group_name == null ? local.unique_identifier : null
+  existing_resource_group_name = var.resource_group_name
+}
 
 # create COS instance for WatsonX.AI project
-# module "cos" {
-#   source            = "terraform-ibm-modules/cos/ibm//modules/fscloud"
-#   version           = "7.5.3"
-#   resource_group_id = module.resource_group.resource_group_id
-#   cos_instance_name = "gen-ai-rag-sample-app-cos-instance"
-#   cos_plan          = "standard"
-# }
+module "cos" {
+  source            = "terraform-ibm-modules/cos/ibm//modules/fscloud"
+  version           = "7.5.3"
+  resource_group_id = module.resource_group.resource_group_id
+  cos_instance_name = "gen-ai-rag-sample-app-cos-instance"
+  cos_plan          = "standard"
+}
+
+# create CD service for toolchain use
+resource "ibm_resource_instance" "cd_instance" {
+  name              = "${var.prefix}-cd-instance"
```

Comment:
what happens if an account already has this service? Will this fail? Should we have a variable to opt in / out?

Reply:
I believe the CD service works on a per-resource group basis, not account. It's still a valid point - how would I check for an existing instance?

Reply:
You can't, you would need the user to specify with a boolean if they want to create the service or not